### PR TITLE
feat: ltex-ls-plus

### PR DESCRIPTION
**Pull request by foxyseta**
Added LTeX+, a popular fork of LTeX which, unlike LTeX, is actively maintained. It has over 4600 downloads on the VS Marketplace: https://marketplace.visualstudio.com/items?itemName=ltex-plus.vscode-ltex-plus

### CODE REVIEW

File: lua/lspconfig/configs/ltex_plus.lua [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+do
+  local enabled_keys = {}
+  for _, ft in ipairs(filetypes) do
+    local id = get_language_id({}, ft)
+    if not enabled_keys[id] then
+      enabled_keys[id] = true
+      table.insert(enabled_ids, id)
+    end
+  end
+end
```
[the remark this answers is not in the document]
Comment (foxyseta) [reply lua/lspconfig/configs/ltex_plus.lua:50]: I initially tried to adhere to the `ltex` package's style.
See 9aabcec41f90272f7344600a858693695fe50cb9 and 4f34f6f48d14caf8ae58a4e83fa4f20e5ccd553b.

Shall I apply all changes to the `ltex` package as well (in a separate PR)?

File: lua/lspconfig/configs/ltex_plus.lua [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  bib = 'bibtex',
+  plaintex = 'tex',
+  rnoweb = 'rsweave',
+  rst = 'restructuredtext',
+  tex = 'latex',
+  pandoc = 'markdown',
+  text = 'plaintext',
+}
+
+local filetypes = {
```
Comment (justinmk) [inline lua/lspconfig/configs/ltex_plus.lua:13]: inline this. it helps with docs generation.

File: lua/lspconfig/configs/ltex_plus.lua [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+local function get_language_id(_, filetype)
+  local language_id = language_id_mapping[filetype]
+  if language_id then
+    return language_id
+  else
+    return filetype
+  end
+end
```
[the remark this answers is not in the document]
Comment (foxyseta) [reply lua/lspconfig/configs/ltex_plus.lua:39]: See https://github.com/neovim/nvim-lspconfig/pull/3442#discussion_r1849149175

File: lua/lspconfig/configs/ltex_plus.lua [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+  'rsweave',
+  'tex',
+  'text',
+  'xhtml',
+}
+
+return {
+  default_config = {
+    cmd = { 'ltex-ls-plus' },
+    filetypes = filetypes,
```
Comment (justinmk) [inline lua/lspconfig/configs/ltex_plus.lua:56]: inline this
```suggestion
    filetypes = {
  'bib',
  'gitcommit',
  'markdown',
  'org',
  'plaintex',
  'rst',
  'rnoweb',
  'tex',
  'pandoc',
  'quarto',
  'rmd',
  'context',
  'html',
  'xhtml',
  'mail',
  'text',
},
```

File: lua/lspconfig/configs/ltex_plus.lua [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    root_dir = util.find_git_ancestor,
+    single_file_support = true,
+    get_language_id = get_language_id,
+    settings = {
+      ltex = {
+        enabled = enabled_ids,
```
Comment (justinmk) [inline lua/lspconfig/configs/ltex_plus.lua:62]: inline

**Comment by foxyseta** [comment]
Thanks! Is it better now?

File: lua/lspconfig/configs/ltex_plus.lua [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+To install, download the latest [release](https://github.com/ltex-plus/ltex-ls-plus) and ensure `ltex-ls-plus` is on your path.
+
+This server accepts configuration via the `settings` key.
+
+```lua
+  settings = {
+		ltex = {
+			language = "en-GB",
+		},
+	},
```
Comment (justinmk) [inline lua/lspconfig/configs/ltex_plus.lua:83]: please follow the codebase conventions. take a hint from the other code in this repo.
```suggestion
  settings = {
    ltex = {
      language = "en-GB",
    },
  },
```

Comment (foxyseta) [reply lua/lspconfig/configs/ltex_plus.lua]: Right, sorry!

**Pull request merged**